geonode/upload/api/views.py

```diff
@@ -91,6 +138,45 @@ def upload(self, request, format=None):
         if not user or not user.is_authenticated:
             return Response(status=status.HTTP_401_UNAUTHORIZED)
 
+        next_step = None
+        max_steps = get_max_amount_of_steps()
+        for n in range(max_steps):
```

Comment:
Is this working with `ASYNC = True` also?

Reply:
Yes.